core/app/[locale]/(default)/checkout/route.ts

```diff
@@ -25,9 +25,12 @@ const CheckoutRedirectMutation = graphql(`
   }
 `);
 
-export async function GET(_: NextRequest, { params }: { params: Promise<{ locale: string }> }) {
+export async function GET(req: NextRequest, { params }: { params: Promise<{ locale: string }> }) {
   const { locale } = await params;
-  const cartId = await getCartId();
+  const url = new URL(req.url);
+  const searchParams = url.searchParams;
+  const paramsCartId = searchParams.get('cartId'); 
+  const cartId = paramsCartId || await getCartId();
```

Comment:
This can be simplified
```suggestion
  const cartId = req.nextUrl.searchParams.get('cartId') ?? await getCartId();
```